### PR TITLE
Spinel: Define properties for discover scan operation

### PR DESCRIPTION
**Pull request by abtink**
This commit defines new spinel properties related to Thread's
discover scan operation. It also updates the spinel draft
documentation.

### CODE REVIEW

File: doc/spinel-protocol-src/spinel-tech-thread.md [hunk 1]
```diff
@@ -287,5 +287,65 @@ CoAP requests/responses from the NCP's RLOC address. This allows the
...
+
+Octets: | 1  |   1  |    2    |   *n*    |    2    |   *n*
+--------|----|------|---------|----------|---------|----------
+Fields: | CH | RSSI | MAC_LEN | MAC_DATA | NET_LEN | NET_DATA
+
+Discover result have two embedded structures which contain
+information about the MAC layer and the NET layer.
+The format below is for discover result:
+
+* `C`: Channel
```
Comment (darconeous) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: I could have sworn we discussed a different solution, which was adding a new scanning mode. Why the change?

Comment (darconeous) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: Maybe I'm confusing myself... I see you did add `SPINEL_SCAN_STATE_DISCOVER`... But how are these properties different than those from the other scan beacon property?

Comment (abtink) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: It's the same approach that we talked about. We add a new scanning mode `SPINEL_SCAN_STATE_DISCOVER` which starts the discover scan operation. The discover operation uses/shares the parameter `SPINEL_PROP_MAC_SCAN_MASK` to decide the channel mask and also uses the newly added `PROP_THREAD_DISOVERY_REQUEST_JOINER_FLAG` and `PROP_THREAD_DISCOVERY_SCAN_ENABLE_FILTERING` as input argument to configure the discover scan operation.

Added the property `PROP_THREAD_DISCOVER_SCAN_RESULT` to emit the discovery scan result. The data items in this property are almost the same as `SPINEL_PROP_MAC_SCAN_BEACON` but with discover result we also have steering data as part of net layer properties which does not really apply in case of active scan.

The key question is if we want to reuse the `SCAN_BEACON` for both active and discover scan result or define a new one like `PROP_THREAD_DISCOVER_SCAN_RESULT`. Both ways seem ok to me. Slight upside of defining a new property is that it can help us distinguish the result and we can in future add new sub-property items to either one independently.

Comment (darconeous) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: Why shouldn't the steering data be a part of the active scan results?

Comment (jwhui) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: Steering data should be passed up as part of active scan results.  We even have a long-standing issue for this #747.

Comment (abtink) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: Synced with RQ @darconeous  offlin... decided to use the `SPINEL_PROP_MAC_SCAN_BEACON` for both active scan and discover scan (and add steering data to it)...
Will change this later which in process will also resolve https://github.com/openthread/openthread/issues/747)

**Comment by abtink** [comment]
In the new push, updated the definitions in `spinel.h` and documentation in spinel draft.

File: doc/spinel-protocol-src/spinel-prop-mac.md [hunk 2]
```diff
@@ -10,15 +10,20 @@ Possible Values:
...
 Set to `SCAN_STATE_ENERGY` to start an energy scan.
-Channel energy will be reported by alternating emissions
-of `PROP_PHY_CHAN` and `PROP_PHY_RSSI`.
+Channel energy result will be reported by emissions
+of `PROP_MAC_ENERGY_SCAN_RESULT` (per channel).
 
-Values switches to `SCAN_STATE_IDLE` when scan is complete.
+Set to `SCAN_STATE_DISOVER` to start a Thread MLE discovery
+scan operation. Discovery scan result will be emitted from
+`PROP_MAC_SCAN_BEACON`.
```
Comment (darconeous) [inline doc/spinel-protocol-src/spinel-prop-mac.md]: **For the future**: We will need to eventually move all thread-specific parts to the thread-specific document. Perhaps we could define a "technology-specific" scan type or maybe a range of types reserved to be defined by the network technology in use.

Comment (abtink) [inline doc/spinel-protocol-src/spinel-prop-mac.md]: :+1:  for the range of types model for specific technology...

File: doc/spinel-protocol-src/spinel-tech-thread.md [hunk 3]
```diff
@@ -287,5 +287,40 @@ CoAP requests/responses from the NCP's RLOC address. This allows the
...
+
+This property is used to enable/disable EUI64 filtering during discovery
+scan operation. Default value is `false`.
+
+### PROP 5397: SPINEL_PROP_THREAD_DISCOVERY_SCAN_PANID (#prop-thread-discovery-scan-panid)
+
+* Type: Read-write
+* Packed-Encoding:: `S`
+
+This property specified the PANID used for filtering during discovery
```
Comment (jwhui) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: nit: `specified` -> `specifies`
